Assignment 2.Rmd

```diff
@@ -75,26 +87,161 @@ plot(g,layout=layout.fruchterman.reingold)
+
+synthesize into EDGE2
+
+
+```{r}
+
+TMAG <- EDGE%>%
+  group_by(from)%>%
+  summarize(actions=sum(count))
+smag <- EDGE%>%
+  group_by(to)%>%
+  summarize(actions=sum(count))
+
+names(TMAG)<-c("ID", "actions")
+names(smag)<-c("ID", "actions")
+
+
+EDGE2 <- bind_rows(TMAG,smag)       #creates a single DF of student and teacher total behaviors
+
+g <- graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX)     #generates the list plot attributes
+
+V(g)$size<-EDGE2$actions*2     #gives the attribute of number of actions to the size of each node
+
+VERTEX<-VERTEX%>%                 #sets the gender
+  mutate(
+    gen.color=as.factor(
+      ifelse(gender=="female","blue","red"))
+    )              
+
+
+#We can change the thickness of the edge according to the number of times a particular teacher has sent a particular student to the principal.
+
+plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gen.color)     #plots g, the network data
+
+
+```
 
 
 ## Part III
 
 Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
 
-Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and dregree. **Who is the most central person in the network?**
+Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and degree. **Who is the most central person in the network?**
+
+```{r}
+library(igraph)
+library(tidyr)
+library(dplyr)
+
+cd<-getwd()
+setwd(cd)
+
+cdf <- read.csv("~/tmp/assignment2/hudk4050-classes.csv", stringsAsFactors=FALSE, na.strings = c("", NA))
+
+cdf$Name<-paste(cdf$First.Name, cdf$Last.Name, sep = " ")       #pastes names together for easy use
+
+cdf2<-cdf%>%                                                    #creates a new DF with desired date
+  gather(key = "Class", value = "ClassX", 3:8, na.rm=TRUE)%>%   #combines all classes into a column
+  select(Name, Class=ClassX)%>%                                 #selects only the classes and user names
+  mutate(Present=1) %>% #attaches a value to a student being in a class
+  filter(Name != "ZIMO CHEN")
+#clean up data using gsub
+#keep data in cdf2
+
+cleaner <- function(arg) {
+  # Function to produce 1 if arg
+  #  contains ’Alice’
+  gsub(pattern = " ",replacement = "", x = arg, ignore.case = FALSE, perl = FALSE)
+}
+cdf2a <- as.data.frame(sapply(X=cdf2, FUN=cleaner), stringsAsFactors = FALSE)
+cleaner1 <- function(arg) {
+  # Function to produce 1 if arg
+  #  contains ’Alice’
+  gsub(pattern = "GR",replacement = "", x = arg, ignore.case = FALSE, perl = FALSE)
+}
```

Comment:
You can also do the following: `cdf2a$Class=gsub(pattern = "GR", replacement = "", cdf2a$Class)`. You don't have to create a function for this necessarily.